test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -577,6 +578,21 @@ public final void before() {
         }
     }
 
+    private final List<CircuitBreaker> breakers = Collections.synchronizedList(new ArrayList<>());
+
+    protected final CircuitBreaker newLimitedBreaker(ByteSizeValue max) {
```

Comment:
I believe this can be reused in other usecases. Specially things that don't use BigArrays